htlcswitch/switch.go

```diff
@@ -785,10 +826,28 @@ func (s *Switch) getLocalLink(pkt *htlcPacket, htlc *lnwire.UpdateAddHTLC) (
 	// Try to find links by node destination.
 	s.indexMtx.RLock()
 	link, err := s.getLinkByShortID(pkt.outgoingChanID)
-	s.indexMtx.RUnlock()
+	defer s.indexMtx.RUnlock()
 	if err != nil {
-		log.Errorf("Link %v not found", pkt.outgoingChanID)
-		return nil, NewLinkError(&lnwire.FailUnknownNextPeer{})
+		// If the link was not found for the outgoingChanID, an outside
+		// subsystem may be using the confirmed SCID of a zero-conf
+		// channel. In this case, we'll consult the Switch maps to see
+		// if an alias exists and use the alias to lookup the link.
+		// This extra step is a consequence of not updating the Switch
+		// forwardingIndex when a zero-conf channel is confirmed. We
```

Comment:
> This extra step is a consequence of not updating the Switch // forwardingIndex when a zero-conf channel is confirme

But this'll be corrected on subsequent restarts after the confirmation?

Reply:
No it won't be corrected, it's because the router will use the confirmed SCID instead of the alias

Reply:
didn't understand comment initially - correcting this would require not populating `forwardingIndex` with `link.ShortChannelID()`, I'm not sure that it's worth it to have an if/else there